### DISCUSSION TITLE
Can BaseURL and others be hard-coded?

### DISCUSSION THREAD

**Discussion by kiokuless** [projectdiscovery/nuclei#4372] [Q&A] [answered]
```
id: example-com-user-id-check

info:
  name: Check 'id' in Response
  author: kiokuless
  severity: info
  tags: test

requests:
  - raw:
      - |+
        GET /user/0 HTTP/1.1
        Host: http://example.com

    matchers:
      - type: word
        words:
          - "id"
        part: body

```

In the template above, {{BaseURL}} is hard-coded as example.com, am I right in understanding that this kind of hard-coding cannot be used?

**Comment by MetzinAround** [comment]
I don't think so, since {{BaseURL}} comes from the target being input, I don't think you can hardcode it into the template. I'd defer to @ehsandeep though.

**Comment by ehsandeep** [comment, marked as answer]
@kiokuless that's possible using `self-contained` template.

See examples : https://github.com/search?q=repo%3Aprojectdiscovery%2Fnuclei-templates%20self-contained&type=code

**Reply by kiokuless** [reply]
I confirmed using `self-contained: true`saved my day! Thank you!

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
